UT4MasterServer.Xmpp/Stanzas/StanzaMessage.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Xml;
+
+namespace UT4MasterServer.Xmpp.Stanzas;
+
+public class StanzaMessage : Stanza
+{
+	public enum TypeAttributeValues
+	{
+		Chat,
+		GroupChat,
+		Headline,
+		Normal
+	}
+
+	public TypeAttributeValues Type { get; set; } = TypeAttributeValues.Normal;
+	public string Body { get; set; } = string.Empty;
+
+	public async Task WriteAsync(XmppWriter writer, CancellationToken cancellationToken)
+	{
+		await Task.Yield();
+
+		writer.OpenTag("message");
+		if (!string.IsNullOrWhiteSpace(ID))
+		{
+			writer.Attribute("id", ID);
+		}
+		writer.Attribute("from", From.Full);
+		writer.Attribute("to", To.Full);
+		if (Error == null)
+		{
+			if (Type != TypeAttributeValues.Normal)
+			{
+				writer.Attribute("type", Type.ToString().ToLower());
+			}
+
```

Comment:
If here supposed to be `else` or what?

Reply:
no, i just tried to highlight indentation in generated xml 😅